### PR TITLE
linuxPackages.nvidiaPackages.legacy_535: 535.154.05 -> 535.216.01

### PR DESCRIPTION
**Pull request by Kiskae**
Prompted by https://github.com/NixOS/nixpkgs/issues/352445#issuecomment-2448313592 (patches a security issue)

Release notes: https://www.nvidia.com/en-us/drivers/details/233000/

@doronbehar Since I can see this branch was added to work around some broken behavior, could you verify that this update doesn't break it?

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Kiskae** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352468`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy535.lib32</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 64 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_6_11.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_6_11.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_6_11.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_6_11.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy535 (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy535)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy535.bin (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy535.bin)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy535.firmware (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy535.firmware)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy535.lib32 (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy535.lib32)</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy535.lib32</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy535 (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy535)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy535.bin (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy535.bin)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy535.firmware (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy535.firmware)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy535.lib32 (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy535.lib32)</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy535</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy535.bin</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy535.firmware</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy535.lib32</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
Thanks for the patch! I will update my system in a few days with this patch included, and after I'll live with it for a few days we can merge this. Do you already run your system with this update?

**Comment by Kiskae** [comment]
> Thanks for the patch! I will update my system in a few days with this patch included, and after I'll live with it for a few days we can merge this. Do you already run your system with this update?

I'm on the unstable 565 beta. I just missed that this variant was added and didn't create PRs for the updates.

**Comment by doronbehar** [comment]
> > Thanks for the patch! I will update my system in a few days with this patch included, and after I'll live with it for a few days we can merge this. Do you already run your system with this update?
> 
> I'm on the unstable 565 beta. I just missed that this variant was added and didn't create PRs for the updates.

Not sure if the answer is no :smile: , but in anycase I will verify that I still don't experience the related issue after I'll update my system. Will do so after GNOME 47 reaches `nixos-unstable`.. https://nixpk.gs/pr-tracker.html?pr=333911

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -115,14 +115,12 @@ rec {
...
   # https://bbs.archlinux.org/viewtopic.php?pid=2155426#p2155426
   legacy_535 = generic {
-    version = "535.154.05";
-    sha256_64bit = "sha256-fpUGXKprgt6SYRDxSCemGXLrEsIA6GOinp+0eGbqqJg=";
-    sha256_aarch64 = "sha256-G0/GiObf/BZMkzzET8HQjdIcvCSqB1uhsinro2HLK9k=";
-    openSha256 = "sha256-wvRdHguGLxS0mR06P5Qi++pDJBCF8pJ8hr4T8O6TJIo=";
-    settingsSha256 = "sha256-9wqoDEWY4I7weWW05F4igj1Gj9wjHsREFMztfEmqm10=";
-    persistencedSha256 = "sha256-d0Q3Lk80JqkS1B54Mahu2yY/WocOqFFbZVBh+ToGhaE=";
-
-    patches = [ rcu_patch ];
```
Comment (doronbehar) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:125]: Just out of curiosity, how did you decide to remove this patch? Does it simply not applying anymore?

**Review by doronbehar — COMMENTED** [review]
Just updated my system with this update, along with GNOME 47, let's see how it goes :).

File: pkgs/os-specific/linux/nvidia-x11/default.nix (hunk 1 above)
Comment (Kiskae) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:125]: Patches for these drivers are pretty much all for kernel compatibility, and those fixes tend to be included with new driver releases from nvidia. So I default to just removing all patches on update and then adding them back if it breaks the build.

Comment (doronbehar) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:125]: > Patches for these drivers are pretty much all for kernel compatibility, and those fixes tend to be included with new driver releases from nvidia. So I default to just removing all patches on update and then adding them back if it breaks the build.

OK makes sense, but it means you have to test the build of this driver against any kernel branch in Nixpkgs now?

Comment (Kiskae) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:125]: https://github.com/NixOS/nixpkgs/pull/352468#issuecomment-2448543129

Yup

**Pull request merged**